src/main/java/org/akaza/openclinica/service/ImportServiceImpl.java

```diff
@@ -939,7 +942,26 @@ private Object validateStudyEvent(StudyEventDataBean studyEventDataBean, StudySu
                     studyEventDataBean.setStudyEventRepeatKey(String.valueOf(studyEvent.getSampleOrdinal()));
                     return studyEvent;
                 } else {
-                    studyEventDataBean.setStudyEventRepeatKey(String.valueOf(eventOrdinal));
+                	// Repeat Key, if found in xml file, then use it
+                	if (studyEventDataBean.getStudyEventRepeatKey() != null && !studyEventDataBean.getStudyEventRepeatKey().equals("")) {   
+                		eventObject = validateEventRepeatKeyIntNumber(studyEventDataBean.getStudyEventRepeatKey());
+                        if (eventObject instanceof ErrorObj) return eventObject;                                      		              	
+                	}else {
+                		studyEventDataBean.setStudyEventRepeatKey(String.valueOf(eventOrdinal));  
```

Comment:
why are you setting the repeat key in xml to the next event ordinal?

Reply:
if no repeat key, then set the repeat key in xml to the next event ordinal is current existing  logic

Reply:
Got it. thanks